Add onError handling to FirebackTicker Retrofit

This will let us gracefully handle the many network-related errors we're seeing...

Selection from the Play Store:
```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:145)
	at android.app.ActivityThread.main(ActivityThread.java:6938)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
Caused by: rx.exceptions.OnErrorNotImplementedException: Unable to resolve host "thebluealliance.firebaseio.com": No address associated with hostname
	at rx.Observable$27.onError(Observable.java:7535)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 8 more
Caused by: java.net.UnknownHostException: Unable to resolve host "thebluealliance.firebaseio.com": No address associated with hostname
	at java.net.InetAddress.lookupHostByName(InetAddress.java:427)
	at java.net.InetAddress.getAllByNameImpl(InetAddress.java:252)
	at java.net.InetAddress.getAllByName(InetAddress.java:215)
	at com.squareup.okhttp.internal.Network$1.resolveInetAddresses(Network.java:29)
	at com.squareup.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:187)
	at com.squareup.okhttp.internal.http.RouteSelector.next(RouteSelector.java:2156)
	at com.squareup.okhttp.internal.http.HttpEngine.createNextConnection(HttpEngine.java:344)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:2327)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at com.thebluealliance.androidclient.datafeed.APIv2RequestInterceptor.intercept(APIv2RequestInterceptor.java:46)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at retrofit.OkHttpCall.execute(OkHttpCall.java:116)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:111)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:88)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable.unsafeSubscribe(Observable.java:7710)
	at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
```

```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5343)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:907)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:702)
Caused by: rx.exceptions.OnErrorNotImplementedException: Connection closed by peer
	at rx.Observable$27.onError(Observable.java:7535)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 8 more
Caused by: javax.net.ssl.SSLHandshakeException: Connection closed by peer
	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:318)
	at com.squareup.okhttp.Connection.connectTls(Connection.java:235)
	at com.squareup.okhttp.Connection.connectAndSetOwner(Connection.java:14199)
	at com.squareup.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:128)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:2328)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at com.thebluealliance.androidclient.datafeed.APIv2RequestInterceptor.intercept(APIv2RequestInterceptor.java:46)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at retrofit.OkHttpCall.execute(OkHttpCall.java:116)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:111)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:88)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable.unsafeSubscribe(Observable.java:7710)
	at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
	Suppressed: javax.net.ssl.SSLHandshakeException: Connection closed by peer
		... 27 more
```
```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5294)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
Caused by: rx.exceptions.OnErrorNotImplementedException: SSL handshake timed out
	at rx.Observable$27.onError(Observable.java:7535)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 8 more
Caused by: java.net.SocketTimeoutException: SSL handshake timed out
	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:318)
	at com.squareup.okhttp.Connection.connectTls(Connection.java:235)
	at com.squareup.okhttp.Connection.connectAndSetOwner(Connection.java:14199)
	at com.squareup.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:128)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:2328)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at com.thebluealliance.androidclient.datafeed.APIv2RequestInterceptor.intercept(APIv2RequestInterceptor.java:46)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at retrofit.OkHttpCall.execute(OkHttpCall.java:116)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:111)
	at retrofit.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:88)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable$2.call(Observable.java:1162)
	at rx.Observable.unsafeSubscribe(Observable.java:7710)
	at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
```

--- a/android/src/main/java/com/thebluealliance/androidclient/fragments/FirebaseTickerFragment.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/fragments/FirebaseTickerFragment.java
@@ -2,10 +2,7 @@ package com.thebluealliance.androidclient.fragments;
 
 import com.google.gson.JsonObject;
 
-import com.firebase.client.DataSnapshot;
 import com.firebase.client.Firebase;
-import com.firebase.client.FirebaseError;
-import com.firebase.client.ValueEventListener;
 import com.thebluealliance.androidclient.Constants;
 import com.thebluealliance.androidclient.R;
 import com.thebluealliance.androidclient.Utilities;
@@ -134,6 +131,13 @@ public abstract class FirebaseTickerFragment extends Fragment implements Action1
                             mChildNodeState = FirebaseChildNodesState.NO_CHILDREN;
                         }
                     }
+                    updateViewVisibility();
+                }, throwable -> {
+                    Log.e(Constants.LOG_TAG, "Firebase rest error: " + throwable);
+                    throwable.printStackTrace();
+
+                    // net error getting item count, show no data view
+                    mChildNodeState = FirebaseChildNodesState.NO_CHILDREN;
                     updateViewVisibility();
                 });
 

